default/python/AI/colonization/calculate_stability.py

```diff
@@ -158,15 +159,20 @@ def _evaluate_xenophobia(planet, species) -> float:
     if AIDependencies.Tags.XENOPHOBIC not in species.tags:
         return 0.0
     colonies = get_colonized_planets()
-    relevant_systems = within_n_jumps(planet.systemID, 5) & set(colonies.keys())
-    result = 0.0
     universe = fo.getUniverse()
-    value = get_named_real("XENOPHOBIC_SELF_TARGET_HAPPINESS_COUNT")
+    empire = fo.getEmpire()
+    max_jumps = get_named_real("XENOPHOBIC_MAX_JUMPS")
+    relevant_systems = within_n_jumps(planet.systemID, max_jumps) & set(colonies.keys())
+    # Do not count owned, different species planets when Racial Purity is adopted
+    if PolicyAI.racial_purity in empire.adoptedPolicies:
+        relevant_systems = relevant_systems.difference(get_owned_planets())
+    penalty_perjump = get_named_real("XENOPHOBIC_TARGET_STABILITY_PERJUMP")
+    result = 0.0
     for sys_id in relevant_systems:
         for pid in colonies[sys_id]:
```

Comment:
Finally, if you want all planets, you need...
```
        system = universe.getSystem(sys_id)
        for pid in system.planetIDs:
```
... and the you could exclude own planets. But you may was well ignore racial purity for now, the AI doesn't adopt it yet.

Reply:
Ok, so I ignored the Racial Purity filtering for now, and left a TODO in place.